src/app/(proper_react)/(redesign)/(authenticated)/user/welcome/EnterInfo.tsx

```diff
@@ -193,7 +194,19 @@ export const EnterInfo = ({
       isRequired: true,
       onChange: setLocation,
     },
-  ].filter((userDetail) => userDetail.isRequired || optionalInfoIsEnabled);
+  ].filter((userDetail) => {
+    const optionalInfoExperimentData =
+      experimentData["welcome-scan-optional-info"];
+    const showOptionalInfo =
+      optionalInfoExperimentData.enabled &&
+      ((userDetail.key === "middle_name" &&
+        (optionalInfoExperimentData.variant === "middleName" ||
+          optionalInfoExperimentData.variant === "suffixAndMiddleName")) ||
+        (userDetail.key === "name_suffix" &&
+          (optionalInfoExperimentData.variant === "suffix" ||
+            optionalInfoExperimentData.variant === "suffixAndMiddleName")));
+    return userDetail.isRequired || showOptionalInfo;
```

Comment:
Optional suggestion, to reduce the large number of nested conjunctions/disjunctions - but not actually sure if this is clearer:

```suggestion
    if (userDetail.isRequired) {
      return true;
    }

    const optionalInfoExperimentData =
      experimentData["welcome-scan-optional-info"];
    if(optionalInfoExperimentData.enabled) {
      if (userDetail.key === "middle_name") {
        return  (optionalInfoExperimentData.variant === "middleName" ||
          optionalInfoExperimentData.variant === "suffixAndMiddleName");
      }
      if (userDetail.key === "name_suffix") {
        return  (optionalInfoExperimentData.variant === "suffix" ||
          optionalInfoExperimentData.variant === "suffixAndMiddleName");
      }
    }
    return false;
```

Reply:
I like the suggestion — it is easier to dive in for someone reading it and the early return makes sense as well. Updated in [64d7693](https://github.com/mozilla/blurts-server/pull/5035/commits/64d769351089c3178ba14523359d35856fc4a2e2).